src/sentry/grouping/strategies/hierarchical.py

```diff
@@ -85,7 +85,7 @@ def get_stacktrace_hierarchy(
             id="stacktrace", values=layer, tree_label=tree_label
         )
 
-    if not all_variants:
+    if not all_variants and components:
```

Comment:
nit: to make parsing the condition easier for humans

```suggestion
    if components and not all_variants:
```